toolchain/check/testdata/facet/named_constant.carbon

```diff
@@ -0,0 +1,195 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/testdata/min_prelude/facet_types.carbon
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/facet/named_constant.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/facet/named_constant.carbon
+
+// --- fail_todo_named_constant_in_rewrite.carbon
+library "[[@TEST_NAME]]";
+
+interface I {
+  let X:! type;
+  let Y:! type;
+}
+
+fn Test() {
+  let Constant:! type = ();
+
+  fn F(T:! I where .X = Constant) {}
+
+  fn G(T:! I where .X = Constant) {
+    // TODO: The facet type T in G should match the facet type T in F.
```

Comment:
It looks like the problem here is that we treat `Constant` as a parameter of `F` that we should be deducing, rather than treating it as an enclosing parameter with a fixed value. Broadly I think the issue in these first four tests is that local `let :!` bindings don't really work yet, rather than anything to do with facet types.